### ISSUE TITLE
return_source_documents does not work for ConversationalRetrievalChain

### ISSUE THREAD

**Issue by Teekyy** [langchain-ai/langchain#16323]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.

### Example Code

The following code initializes the chatbot instance using ConversationalRetrievalChain with the 'return_source_documents' parameter: 
```
def initialize_chatbot(index_name):
    chatbot = ChatOpenAI(
        openai_api_key=os.environ["OPENAI_API_KEY"],
        model='gpt-3.5-turbo',
        temperature=0.2
    )

    embeddings = OpenAIEmbeddings(openai_api_key=os.environ["OPENAI_API_KEY"])
    vectorstore = Pinecone.from_existing_index(index_name, embeddings)
    retriever = vectorstore.as_retriever()

    memory = ConversationBufferWindowMemory(
        k=10,
        memory_key="chat_history",
        return_messages=True
    )

    qa = ConversationalRetrievalChain.from_llm(
        llm=chatbot, 
        retriever=retriever, 
        memory=memory,
        return_source_documents=True
    )

    return qa
 ```

The following code runs a query:
```
def chat(query, qa):
    response = qa(query)

    print(response)

query = "what is the nutrition information for the boiled egg recipe?"
chat(query, chatbot)
```

The error I get:
```
File [langchain/chains/base.py:314], in Chain.__call__(self, inputs, return_only_outputs, callbacks, tags, metadata, run_name, include_run_info)
    [312]     raise e
    [313]run_manager.on_chain_end(outputs)
--> [314] final_outputs: Dict[str, Any] = self.prep_outputs(
    [315]    inputs, outputs, return_only_outputs
    [316] )
    [317] if include_run_info:
    [318]     final_outputs[RUN_KEY] = RunInfo(run_id=run_manager.run_id)

File [langchain/chains/base.py:410], in Chain.prep_outputs(self, inputs, outputs, return_only_outputs)
    [408] self._validate_outputs(outputs)
    [409] if self.memory is not None:
--> [410]     self.memory.save_context(inputs, outputs)
    [411] if return_only_outputs:
...
---> [29]         raise ValueError(f"One output key expected, got {outputs.keys()}")
     [30]     output_key = list(outputs.keys())[0]
     [31] else:

ValueError: One output key expected, got dict_keys(['answer', 'source_documents'])
```

### Description

I am trying to use the langchain library to return source documents using the ConversationalRetrievalChain. However I keep getting an error relating to it expecting only one output key. I looked into the code and found I think it is executing the `__call__` function, which was deprecated in langchain version 0.1.0, and it says it only expects one output key. I am using the most recent langchain version that pip allows (`pip install --upgrade langchain`), which is 0.1.1. How can I get this to execute properly?

Additional notes:
- I am using langchain-openai for ChatOpenAI and OpenAIEmbeddings

### System Info

"pip install --upgrade langchain"
Python 3.11.5
Langchain 1.1.0

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [X] Chains
- [ ] Callbacks/Tracing
- [ ] Async

**Comment by Teekyy** [comment]
Obviously in this case I want the source documents, so making that parameter false would not be ideal.

**Comment by Teekyy** [comment]
Upon further investigation, if you remove the `memory` parameter, and instead change the code to run a query to `response = qa({"question": query, "chat_history": ""})`, it seems to work. Seems like its messing with the memory component, but not sure why.

**Comment by Teekyy** [comment]
Solved by this issue: #6741
